File: LC/python/Plus One.py
class Solution:
    # @param digits, a list of integer digits
    # @return a list of integer digits
    def plusOne(self, digits):
        (digits[-1],full) = self.addone(digits[-1]+1,0)
        for i in range(len(digits)-2,-1,-1):
            (digits[i],full) = self.addone(digits[i],full)
        if full == 1:
            digits.insert(0,1)
        return digits
        
    def addone(self,digit,full):
        res = digit + full
        digit = res % 10
        full = res / 10
        return (digit, full)
        

File: LC/java/Plus One.java
public class Solution {
    public int[] plusOne(int[] digits) {
        int flag = 1,tmp;
        
        for(int i = digits.length-1; i >= 0; i--){
            tmp = digits[i]+flag;
            digits[i] = tmp % 10;
            flag = tmp / 10;
        }
        
        if(flag != 0){
            int[] res = new int[digits.length+1];
            res[0] = flag;
            for(int i = 0; i < digits.length; i++){
                res[i+1] = digits[i];
            }
            return res;
        }
        return digits;
    }
}